File: Level_4(C++,Java)/C++/static/Identifiable/Identifiable.cpp
#include "Identifiable.h"

int Identifiable::lastId = 0;

Identifiable::Identifiable() {
    lastId += 1;
    this->id = lastId;
}

Identifiable::Identifiable(const Identifiable& copy) {
    lastId += 1;
    this->id = lastId;
}

Identifiable::~Identifiable() {}

int Identifiable::getId() const {
    return this->id;
}

Identifiable& Identifiable::operator=(const Identifiable& copy) {
    return *this;
}

File: Level_4(C++,Java)/java/Static/Identifiable/Identifiable.java
public class Identifiable {
    private int id;
    private static int lastId = 0;
    
    public Identifiable() {
        lastId += 1;
        this.id = lastId;
    }
    
    public Identifiable(Identifiable copy) {
        lastId += 1;
        this.id = lastId;
    }
    
    public int getId() {
        return this.id;
    }
}
